src/schematools/utils.py

```diff
@@ -109,7 +124,7 @@ def def_from_url(
         response = connection.get(base_url / index[dataset_id])
         response.raise_for_status()
 
-        dataset_schema: types.ST = data_type.from_dict(response.json())
+        dataset_schema: types.SchemaType = data_type.from_dict(response.json())
```

Comment:
`ST` was correct. By downcasting every possible `SchemaType` to `SchemaType` here, you'll need upcasts to more specific types elsewhere. `ST`'s definition prevents that.